### ISSUE TITLE
Problems compiling (Cannot open module file 'module_wrf_top.mod')

### ISSUE THREAD

**Issue by Cesar-IIUNAM** [jcwarner-usgs/COAWST#98]
Hi there,

When I try to compile the Sandy test project or even any project that includes the WRF, I get this error:

cd ./Build; /usr/bin/mpif90 -c -frepack-arrays -fallow-argument-mismatch -O3 -ffast-math -O3 -ftree-vectorize -ftree-loop-linear -funroll-loops -w -ffree-form -ffree-line-length-none -frecord-marker=4 -fconvert=big-endian -I/usr/include -I/usr/local/mct/include -I/home/ctorresm/Desktop/COAWST///WRF/main -I/home/ctorresm/Desktop/COAWST///WRF/external/esmf_time_f90 -I/home/ctorresm/Desktop/COAWST///WRF/frame -I/home/ctorresm/Desktop/COAWST///WRF/share master.f90
f951: Warning: Nonexistent include directory ‘/usr/local/mct/include’ [-Wmissing-include-dirs]
master.f90:26:11:

   26 |       USE module_wrf_top, ONLY : wrf_init
      |           1
Fatal Error: Cannot open module file ‘module_wrf_top.mod’ for reading at (1): No such file or directory
compilation terminated.
make: *** [Master/Module.mk:25: Build/master.o] Error 1

I could not find the module_wrf_top.mod file. In another project I had a similar problem when compiling, where the program could not find a xxx.mod file, I solved it by copying the .mod files to /usr/include.

I would be very grateful if you could give me some guidance on what could be happening. 

Best regards,
César

**Comment by jcwarner-usgs** [comment]
you need to compile MCT.  we distribute that code as part of the COAWST system.
then set incdir and libdir, as described on page 6 in the Manual.
let me know if that works for you.

**Comment by Cesar-IIUNAM** [comment]
Hi JC,
Thanks for your support. The problem persists. In fact I had already compiled MCT, I compiled again anyway. 
I don't understand why they are stored in /usr/local/include and /usr/local/lib.
However, I have set the paths like this (I get the same error):

MCT_INCDIR=/usr/local/include
MCT_LIBDIR=/usr/local/lib

**Comment by jcwarner-usgs** [comment]
when you install MCT, where did you put the libs and where did you put the mods?
you need to set the MCT incdir and libdir paths to the locations of the mods and the libs.
if libmct and libmpeu are not in /usr/local/lib, then dont point to that.
you need to set MCT_libdir to be the location where you put the libs.

**Comment by Cesar-IIUNAM** [comment]
When I install MCT, the libs are stored in:
/usr/local/include
/usr/local/lib
I checked that libmct and libmpeu are in /usr/local/lib, like this:  
![image](https://user-images.githubusercontent.com/113042121/192535620-a93064c8-3e9d-488d-aa39-03af13b51172.png)


Also, i have set the paths as follows (see snapshot), is it ok?
MCT_INCDIR=/usr/local/include
MCT_LIBDIR=/usr/local/lib

![image](https://user-images.githubusercontent.com/113042121/192536006-56e7bb8c-31e2-48f3-94c1-1f23569b3969.png)

**Comment by jcwarner-usgs** [comment]
at the command prompt, what do you get if you type

$MCT_INCDIR

**Comment by Cesar-IIUNAM** [comment]
MCT_INCDIR: command not found

**Comment by jcwarner-usgs** [comment]
so i dont think you successfully set that in your env. type 
env
and see if you find any mct stuff.

**Comment by Cesar-IIUNAM** [comment]
My apologies JC, I made a mistake earlier. When I enter in my command prompt: echo $MCT_INCDIR I get the following: /usr/local/include
When I type echo $"MCT_INCDIR" I get the following: MCT_INCDIR
It looks like the environment variable is set correctly

**Comment by jcwarner-usgs** [comment]
try this exactly:
[jwarner@poseidon-l1 ccbay46]$    $MCT_INCDIR

and here is what i get:
-bash: /vortexfs1/usgs/jwarner/models/MCT_stack/include: Is a directory

it should tell you the path to the libs

**Comment by Cesar-IIUNAM** [comment]
Well, I get this:

ctorresm@ctorresm-VirtualBox:~$ $MCT_INCDIR
bash: /usr/local/include: Is a directory

**Comment by jcwarner-usgs** [comment]
ok.  that is what it should say, if that is what you set.
I am not sure what is going on, the compilation looks for the mct incdir.
try this
edit Compilers/Linux-gfortran.mk
and change lines 232-237 from
ifdef USE_MCT
       MCT_INCDIR ?= /usr/local/mct/include
       MCT_LIBDIR ?= /usr/local/mct/lib
           FFLAGS += -I$(MCT_INCDIR)
             LIBS += -L$(MCT_LIBDIR) -lmct -lmpeu
endif

to

ifdef USE_MCT
       MCT_INCDIR ?= /usr/local/include
       MCT_LIBDIR ?= /usr/local/lib
           FFLAGS += -I$(MCT_INCDIR)
             LIBS += -L$(MCT_LIBDIR) -lmct -lmpeu
endif

**Comment by Cesar-IIUNAM** [comment]
For a moment I thought that compilation would be successful, because I was not getting the warning that /usr/local/mct/include was not found.

I get the same error, should the file module_wrf_top.mod be in the include folder?

![image](https://user-images.githubusercontent.com/113042121/192663230-148422c8-1ceb-4244-bc9a-10b6d3bc0e2b.png)

**Comment by jcwarner-usgs** [comment]
that happens sometimes because the build failed earlier. do this.

scrip build.log
./coaws.bash -j 4
exit

that will create a file called buld.log, send that file to me.

**Comment by Cesar-IIUNAM** [comment]
Thanks for your support. I send the build.log file
[build.log](https://github.com/jcwarner-usgs/COAWST/files/9669075/build.log)

**Comment by jcwarner-usgs** [comment]
near the top of the log i see
 
Compiling wrf
/bin/sh: 6: ./compile: not found
-------- Finished compiling WRF ------------

that is not good.
cd /home/ctorresm/Desktop/COAWST///WRF
and do 
ls -ltr
what does it show?
is there a file called   compile   ?

**Comment by Cesar-IIUNAM** [comment]
Indeed, there is a file called "compile".

total 248
-rwxrwxr-x  1 ctorresm ctorresm  1027 sep 12 09:14 LICENSE.txt
-rwxrwxr-x  1 ctorresm ctorresm  1143 sep 12 09:14 README.md
-rwxrwxr-x  1 ctorresm ctorresm 17648 sep 12 09:14 README
-rwxrwxr-x  1 ctorresm ctorresm 60005 sep 12 09:14 Makefile
drwxrwxr-x  2 ctorresm ctorresm  4096 sep 12 09:14 Registry
drwxrwxr-x  2 ctorresm ctorresm  4096 sep 12 09:14 arch
drwxrwxr-x  3 ctorresm ctorresm 12288 sep 12 09:14 chem
-rwxrwxr-x  1 ctorresm ctorresm 37163 sep 12 09:14 configure
**-rwxrwxr-x  1 ctorresm ctorresm 20697 sep 12 09:14 compile**
-rwxrwxr-x  1 ctorresm ctorresm  3385 sep 12 09:14 clean
drwxrwxr-x  2 ctorresm ctorresm  4096 sep 12 09:14 doc
drwxrwxr-x  2 ctorresm ctorresm  4096 sep 12 09:14 dyn_em
drwxrwxr-x  2 ctorresm ctorresm  4096 sep 12 09:14 dyn_nmm
drwxrwxr-x 16 ctorresm ctorresm  4096 sep 12 09:14 external
drwxrwxr-x  2 ctorresm ctorresm  4096 sep 12 09:14 frame
drwxrwxr-x 17 ctorresm ctorresm  4096 sep 12 09:14 hydro
drwxrwxr-x  2 ctorresm ctorresm  4096 sep 12 09:14 inc
drwxrwxr-x 15 ctorresm ctorresm  4096 sep 12 09:14 hydro_v5.0
drwxrwxr-x  2 ctorresm ctorresm  4096 sep 12 09:14 main
drwxrwxr-x  2 ctorresm ctorresm 12288 sep 12 09:14 phys
drwxrwxr-x  2 ctorresm ctorresm  4096 sep 12 09:15 run
drwxrwxr-x  2 ctorresm ctorresm  4096 sep 12 09:15 share
drwxrwxr-x 19 ctorresm ctorresm  4096 sep 12 09:15 test
drwxrwxr-x  4 ctorresm ctorresm  4096 sep 12 09:15 tools
drwxrwxr-x 13 ctorresm ctorresm  4096 sep 12 09:15 var
drwxrwxr-x  2 ctorresm ctorresm  4096 sep 12 09:15 wrftladj

**Comment by jcwarner-usgs** [comment]
something is not correct.  i need you to try and figure it out.

Maybe the extra '//' in the path is giving troubles???
this 
cd /home/ctorresm/Desktop/COAWST///WRF
should be
cd /home/ctorresm/Desktop/COAWST/WRF

if you cd to that location and type.
./compile
what does it do? 
try to remove those extra //

**Comment by Cesar-IIUNAM** [comment]
I am trying to solve it.
Apparently the problem is when I try to compile the models coupled with WRF. When it's ROMS-SWAN, the compilation is successful. 
Honestly, I don't know why the extra //. 
I don't think it's because I'm using a virtual machine. 

When cd to /home/ctorresm/Desktop/COAWST///WRF
and then ./compile
I get this:

ctorresm@ctorresm-VirtualBox:~/Desktop/COAWST$ cd /home/ctorresm/Desktop/COAWST///WRF
ctorresm@ctorresm-VirtualBox:~/Desktop/COAWST/WRF$ ./compile
bash: ./compile: /bin/csh: bad interpreter: No such file or directory
ctorresm@ctorresm-VirtualBox:~/Desktop/COAWST/WRF$

**Comment by jcwarner-usgs** [comment]
so the "compile" file is actually just a text file. it starts with

#!/bin/csh -f
setenv START_OF_COMPILE "`date`"
...

so maybe your system does not know what to do with c shell (csh).
try this, type
ls -ltr /bin/csh

I get: 
lrwxrwxrwx 1 root root 4 Mar 12  2018 /bin/csh -> tcsh

**Comment by Cesar-IIUNAM** [comment]
well, i get this:

ctorresm@ctorresm-VirtualBox:~/Desktop$ ls -ltr /bin/csh
ls: cannot access '/bin/csh': No such file or directory

**Comment by jcwarner-usgs** [comment]
WRF requires the csh to work.  you will have to get that shell installed.

**Comment by jcwarner-usgs** [comment]
did you fix this yet?  i would like to close this ticket if it is you needing to install csh.

**Comment by Cesar-IIUNAM** [comment]
I have installed csh.
ctorresm@ctorresm-VirtualBox:~/Desktop/COAWST$ ls -ltr /bin/csh
lrwxrwxrwx 1 root root 21 oct 13 15:19 /bin/csh -> /etc/alternatives/csh

However, for some reason I keep having the same problem compiling the Sandy project. 

I/usr/local/include -I/home/ctorresm/Desktop/COAWST///WRF/main -I/home/ctorresm/Desktop/COAWST///WRF/external/esmf_time_f90 -I/home/ctorresm/Desktop/COAWST///WRF/frame -I/home/ctorresm/Desktop/COAWST///WRF/share master.f90
master.f90:26:11:

What folder should module_wrf_top.mod be in? I can't find it.

   26 |       USE module_wrf_top, ONLY : wrf_init
      |           1
Fatal Error: Cannot open module file ‘module_wrf_top.mod’ for reading at (1): No such file or directory
compilation terminated.
make: *** [Master/Module.mk:25: Build/master.o] Error 1

**Comment by jcwarner-usgs** [comment]
you need to look at the whole std out.  if an error happened way at the top, wrf will keep compiling and the error you see there is because of something that started earlier.

**Comment by iarzeno7** [comment]
Hello, I am running into the same issue that was posted here (except with the coupled JOE_TC test case). Was this ever resolved? I have successfully compiled and run the coupled Inlet_test case. I have also read through all the previous comments in this thread in an attempt to resolve this myself. I'm attaching my own build.log. Any insight would be greatly appreciated. 

[build.log](https://github.com/jcwarner-usgs/COAWST/files/9883800/build.log)

**Comment by jcwarner-usgs** [comment]
so you have a different problem.  the previous user did not have csh in their system, then i am not sure what happened because they did not include the full build.
When WRF compiles, if it gets an error, it keeps going. So then at the bottom when the build is pulling libs together, it cant find something.  But you need to go to the top of the output and start from there, not at the bottom.
For you i see:

Compiling wrf
You must run the 'configure' script before running the 'compile' script!
Exiting...

I dont think it even started to compile wrf.  I also see:

NETCDF4 IO features are requested, but this installation of NetCDF           
  /share/software/user/open/netcdf-fortran/4.5.4
DOES NOT support these IO features.                                          

Please make sure NETCDF version is 4.1.3 or later and was built with         
--enable-netcdf4                                                             

OR set NETCDF_classic variable                                               
   bash/ksh : export NETCDF_classic=1
      echo  csh : setenv NETCDF_classic 1

Then re-run this configure script                                            

!!! configure.wrf has been REMOVED !!!

after that it did not compile wrf. So ....
can you check if your installation of netcdf has --enable-netcdf4?

-j

**Comment by iarzeno7** [comment]
Hi John,

Thank you for pointing this out! I do use a netcdf version that is 4.1.3 or higher, as required, but I don't know if it has --enable-netcdf4, since I did not do the installation myself. (I am using a university server that already had the gfortran netcdf version installed.) I will check with IT.

However, I did add "export NETCDF_classic=1", which helped the compilation. My error now is completely different (I think I'm running out of memory), so I'll mark this particular issue (cannot open module file...) as resolved for me. 

Thank you!
